portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -24,34 +24,20 @@
 import com.google.appengine.api.users.UserServiceFactory;
 import com.google.gson.Gson;
 import com.google.sps.interfaces.DatastoreInterface;
+import com.google.sps.controllers.Authenticator;
+import com.google.sps.data.LoginInfo;
 
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
 
-  private static final String RETURN_URL = "/index.html#comment-section";
   private Gson gson = new Gson();
   private DatastoreInterface datastoreInterface = new DatastoreInterface();
-  
-  private class LoginInfo{
-      private boolean isLoggedIn;
-      private String url;
-      private LoginInfo(UserService userService){
-          this.isLoggedIn = userService.isUserLoggedIn();
-          
-          if(this.isLoggedIn){
-              this.url = userService.createLogoutURL(RETURN_URL);
-          }else{
-              this.url = userService.createLoginURL(RETURN_URL);
-          }
-      }
-
-  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(MediaType.APPLICATION_JSON);
-    UserService userService = UserServiceFactory.getUserService();
-    LoginInfo loginInfo = new LoginInfo(userService);
+    Authenticator authenticator = new Authenticator();
```

Comment:
It probably makes sens to create Authenticator instance just once in the LoginServlet, instead for every get request.

Reply:
This is actually difficult in my opinion because it has a method that is clearly scoped to the current request so maybe it's better to create a new instance for every request because each one could be for a different user. I also don't know exactly what guarantees the UserService gives but i probably makes sense to dig a bit into the docs to find out if you have to create a new instance for every request or not.

Reply:
I tried to looked into these docs I think on Wednesday, but haven't managed to find the documentation I was looking for. However, I think the server creates a different session for each user, and the userservice determines the user based on the session. Thus it would actually work if I only instanciated it once. But as I said, I haven't found it in the docs, how could I search for this? (Or maybe I was looking at the wrong docs.)

Reply:
Docs https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/users/UserService do not mention anything around topic of request scoping unfortunately.
So I guess for safety let's keep it as is.